src/base/ClpMap.h

```diff
@@ -272,6 +280,20 @@ ClpMap<Key, Value, MemoryUsedBy>::del(const Key &key)
         erase(i);
 }
 
+template <class Key, class Value, uint64_t MemoryUsedBy(const Value &)>
+void
+ClpMap<Key, Value, MemoryUsedBy>::merge(const ClpMap<Key, Value, MemoryUsedBy> &other)
+{
+    // reverse-order across other - to preserve old LRU
+    for (auto i = other.entries_.rbegin(); i != other.entries_.rend(); ++i) {
+        if (!i->expired()) {
+            const auto ours = find(i->key);
+            const auto newTtl = (ours != index_.cend() ? max(i->expires, ours->second->expires) : i->expires) - squid_curtime;
```

Comment:
If this merge() method survives, we should try to refactor so that its code does not convert expiration values to TTL values (here) and then back to the original expiration value (in Entry constructor). In other words, there should be no `squid_curtime` involvement; just `expires` manipulation.